src/metabase/models/dashboard.clj

```diff
@@ -517,43 +516,32 @@
       (events/publish-event! :event/card-create {:object card :user-id (:creator_id card)})
       (t2/hydrate card :creator :dashboard_count :can_write :can_run_adhoc_query :collection))))
 
-(defn- ensure-unique-collection-name
-  [collection-name parent-collection-id]
-  (let [c (t2/count :model/Collection
-                    :name     [:like (format "%s%%" collection-name)]
-                    :location (collection/children-location (t2/select-one [:model/Collection :location :id]
-                                                                           :id parent-collection-id)))]
-    (if (zero? c)
-      collection-name
-      (format "%s %s" collection-name (inc c)))))
-
 (defn save-transient-dashboard!
   "Save a denormalized description of `dashboard`."
   [dashboard parent-collection-id]
   (let [{dashcards      :dashcards
          tabs           :tabs
-         dashboard-name :name
          :keys          [description] :as dashboard} (i18n/localized-strings->strings dashboard)
-        collection (xrays/create-collection!
-                    (ensure-unique-collection-name dashboard-name parent-collection-id)
-                    "Automatically generated cards."
-                    parent-collection-id)
         dashboard  (first (t2/insert-returning-instances!
                            :model/Dashboard
                            (-> dashboard
                                (dissoc :dashcards :tabs :rule :related
                                        :transient_name :transient_filters :param_fields :more)
                                (assoc :description description
-                                      :collection_id (:id collection)
-                                      :collection_position 1))))
+                                      :collection_id parent-collection-id))))
         {:keys [old->new-tab-id]} (dashboard-tab/do-update-tabs! (:id dashboard) nil tabs)]
     (add-dashcards! dashboard
                     (for [dashcard dashcards]
-                      (let [card     (some-> dashcard :card (assoc :collection_id (:id collection)) save-card!)
-                            series   (some->> dashcard :series (map (fn [card]
-                                                                      (-> card
-                                                                          (assoc :collection_id (:id collection))
-                                                                          save-card!))))
+                      (let [card     (some-> dashcard :card
+                                             (assoc :dashboard_id (:id dashboard)
+                                                    :collection_id parent-collection-id)
+                                             save-card!)
+                            series   (some->> dashcard
+                                              :series
+                                              (mapv (fn [card]
```

Comment:
I changed this to a `mapv` because side-effects in a `map` make me super nervous.